clients/spark/core/src/test/scala/io/treeverse/jpebble/BlockParserSpec.scala

```diff
@@ -261,3 +266,175 @@ class CountedIteratorSpec extends AnyFunSpec with Matchers {
+            (new String(entry.key), new String(entry.value))).toSeq
+
+          actual should contain theSameElementsInOrderAs expected
+        })
+      }
+    }
+
+    describe("with random restart interval") {
+      it("should parse successfully") {
+        val fileName = "fuzz.block.restart.interval"
+        withGeneratedTestFiles(fileName, (in: BlockReadable, expected: Seq[(String, String)], sstSize: Long) => {
+          val it = BlockParser.entryIterator(in)
+          val actual = it.map((entry) =>
+            (new String(entry.key), new String(entry.value))).toSeq
+
+          actual should contain theSameElementsInOrderAs expected
+        })
+      }
+    }
+
+    describe("with random block size") {
+      it("should parse successfully") {
+        val fileName = "fuzz.block.size"
+        withGeneratedTestFiles(fileName, (in: BlockReadable, expected: Seq[(String, String)], sstSize: Long) => {
+          val it = BlockParser.entryIterator(in)
+          val actual = it.map((entry) =>
+            (new String(entry.key), new String(entry.value))).toSeq
+
+          actual should contain theSameElementsInOrderAs expected
+        })
+      }
+    }
+
+    describe("with random blocksize threshold") {
+      it("should parse successfully") {
+        val fileName = "fuzz.block.size.threshold"
+        withGeneratedTestFiles(fileName, (in: BlockReadable, expected: Seq[(String, String)], sstSize: Long) => {
+          val it = BlockParser.entryIterator(in)
+          val actual = it.map((entry) =>
+            (new String(entry.key), new String(entry.value))).toSeq
+
+          actual should contain theSameElementsInOrderAs expected
+        })
+      }
+    }
+  }
+
+  def copyTestFile(fileName: String, suffix: String) : File =
+    container.copyFileFromContainer("/local/" + fileName + suffix, in => {
+      val tempOutFile = File.createTempFile("test-block-parser.", suffix)
+      tempOutFile.deleteOnExit()
+      val out = new java.io.FileOutputStream(tempOutFile)
+      IOUtils.copy(in, out)
+      in.close()
+      out.close()
+      return tempOutFile
+    })
+
+
+  /**
+   * Lightweight fixture for running particular tests with SSTables and Json fules generated by a go application on
+   * test startup. The fixture uses sstables as the parser input, and the json as the source of the expected output of
+   * the parsing operation.
+   */
+  def withGeneratedTestFiles(filename: String, test: (BlockReadable, Seq[(String, String)], Long) => Any) = {
+    val tmpSstFile = copyTestFile(filename, ".sst")
+    val tmpJsonFile = copyTestFile(filename, ".json")
+
+    val in = new BlockReadableFileChannel(new java.io.FileInputStream(tmpSstFile).getChannel)
+    val jsonString = os.read(os.Path(tmpJsonFile.getAbsolutePath))
+    val data = ujson.read(jsonString)
+    val expected = data.arr.map(e => (e("Key").str, e("Value").str))
+    try {
+      test(in, expected, tmpSstFile.length())
+    } finally {
```

Comment:
Best pull all of these files inside the `try`, I guess, otherwise we can leave a file open for no good reason.